unit-tests/src/test/scala/java/util/regex/PatternSuite.scala

```diff
@@ -576,21 +576,48 @@ object PatternSuite extends tests.Suite {
       }
     )
 
+    /// Ordered alphabetical by description (second arg).
+    /// Helps ensuring that each scalanative/regex Parser description
+    /// matches its JVM equivalent.
+    ///
+    /// These are _not_ all the JVM parser PatternSyntaxExceptions available.
+    /// They are only the cases reported by scalanative.regex.Parser.scala
+    ///
+    /// Some tests are marked MISSING because I (LeeT.) have not yet
+    /// figured out a pattern which will trigger the error.
+
+    // MISSING: Test scalanative.regex ERR_INVALID_NAMED_CAPTURE
+    //          "Bad named capture group"
+
+    // There are two conditions which currently yield the same description.
+    // ERR_MISSING_REPEAT_ARGUMENT matches the JVM.
+    // Trigger for ERR_MISSING_REPEAT_ARGUMENT  not yet found.
+
+    // Test scalanative.regex ERR_MISSING_REPEAT_ARGUMENT
+    syntax("*", "Dangling meta character '*'", 0)
+
+    // MISSING: Test scalanative.regex ERR_MISSING_REPEAT_ARGUMENT
+    //          "Dangling meta character '*'"
+
+    syntax("[a-0]", "Illegal character range", 3)
     syntax("foo\\Lbar", "Illegal/unsupported escape sequence", 4)
-    syntax("foo[bar", "Unclosed character class", 6)
+
+    // MISSING: Test scalanative.regex ERR_INVALID_REPEAT_OP
+    //          "invalid nested repetition operator"
+
     syntax("foo\\", "Trailing Backslash", 4)
-    syntax("[a-0]", "Illegal character range", 3)
-    syntax("*", "Dangling meta character '*'", 0)
-    syntax("foo(bar(foo)baz", "Missing parenthesis", 15)
-    syntax("foo(foo)bar)baz", "Missing parenthesis", 10)
+    syntax("foo[bar", "Unclosed character class", 6)
+    syntax("foo(bar(foo)baz", "Unclosed group", 15)
+    syntax("(?q)", "Unknown inline modifier", 2) // bad perl operator
+    syntax("foo(foo)bar)baz", "Unmatched closing ')'", 10)
   }
 
   private def syntax(pattern: String, description: String, index: Int): Unit = {
     assertThrowsAnd[PatternSyntaxException](Pattern.compile(pattern))(
       e => {
-        e.getDescription == description &&
-        e.getPattern == pattern
-        e.getIndex == index
+        (e.getDescription == description) &&
+        (e.getPattern == pattern) &&
+        (e.getIndex) == (index)
```

Comment:
```suggestion
        (e.getIndex == index)
```
I don't see the point of putting individual properties and vars in parens.

But perhaps none of those new parentheses are needed? In the similar snippet further down, you did not add any parens.

Reply:
Thank you.  Fixed now (was not intended. sloppy on my part...).

I also made the relevant code in scalanative/regex/PatternSuite.scala correspond.

As you noted, the parentheses are not _strictly_ necessary.  I find that some
(proper & useful) parentheses make it easier me read logical statements on the fly.
Perhaps I have read too much Lisp code.